### ISSUE TITLE
Ray Cast from the surface of a rotated Cuboid hits said Cuboid again

### ISSUE THREAD

**Issue by ldesgoui** [rustsim/ncollide#242]
Hello,

Sorry about the vague title, I can't really find a concise way to describe it
Simulating rays bouncing off Shapes (in the case of a ray tracing algorithm) sometimes fails because a rotated Cuboid would "eat" my rays. If not rotated, everything is fine. Testing with f64s produces the same result.

Here is an example test that I think should pass if this issue is fixed (and if I am not using the library in the wrong way)
https://gist.github.com/ldesgoui/7d35c793a7663d6750952bd5d861ee81
In short, this tests that a ray originating from an intersection's point nudged outside a shape should miss that shape 

Thanks

**Comment by sebcrozet** [comment]
I confirm this is a bug in the box ray-cast implementation.
A fix will be published today.
